File: apollo-api/src/main/java/com/apollographql/apollo/api/GraphqlFragment.java
package com.apollographql.apollo.api;

import com.apollographql.apollo.api.internal.ResponseFieldMarshaller;

/**
 * Represents a GraphQL fragment
 */
public interface GraphqlFragment {

  /**
   * Returns marshaller to serialize fragment data
   *
   * @return {@link com.apollographql.apollo.api.internal.ResponseFieldMarshaller} to serialize fragment data
   */
  ResponseFieldMarshaller marshaller();
}

File: apollo-api/src/main/java/com/apollographql/apollo/api/GraphqlFragment.kt
package com.apollographql.apollo.api

import com.apollographql.apollo.api.internal.ResponseFieldMarshaller

/**
 * Represents a GraphQL fragment
 */
interface GraphqlFragment {

  /**
   * Returns marshaller [ResponseFieldMarshaller] to serialize fragment data
   */
  fun marshaller(): ResponseFieldMarshaller
}
